client/lib/domains/is-subdomain.ts

```diff
@@ -0,0 +1,16 @@
+import { getRootDomain } from 'calypso/lib/domains/utils';
+
+export function isSubdomain( domainName: string ) {
```

Comment:
```suggestion
export function isSubdomain( domainName: string ): boolean {
```

Typescript can infer the function's return type, but what do you think of adding this to be extra explicit?